+adi/+internal/StingrayControl.m

```diff
@@ -18,69 +18,50 @@
             end
         end
         
         function set.PowerUpDown(obj, value)
-            obj.setAttributeRAW('voltage5', 'raw', num2str(value), true, obj.SRayCtrlDevices);
+            obj.setAttributeRAW('voltage5', 'raw', num2str(value), true, obj.SRayCtrlDevice);
         end
         
         function result = get.Ctrl5V(obj)
             result = false;
-            if ~isempty(obj.SRayCtrlDevices)
-                result = obj.getAttributeRAW('voltage4', 'raw', true, obj.SRayCtrlDevices);
+            if ~isempty(obj.SRayCtrlDevice)
+                result = obj.getAttributeRAW('voltage4', 'raw', true, obj.SRayCtrlDevice);
             end
         end
         
         function set.Ctrl5V(obj, value)
-            obj.setAttributeRAW('voltage4', 'raw', num2str(value), true, obj.SRayCtrlDevices);
+            obj.setAttributeRAW('voltage4', 'raw', num2str(value), true, obj.SRayCtrlDevice);
         end
         
         function result = get.PAOn(obj)
             result = false;
-            if ~isempty(obj.SRayCtrlDevices)
-                result = obj.getAttributeRAW('voltage0', 'raw', true, obj.SRayCtrlDevices);
+            if ~isempty(obj.SRayCtrlDevice)
+                result = obj.getAttributeRAW('voltage0', 'raw', true, obj.SRayCtrlDevice);
             end
         end
         
         function set.PAOn(obj, value)
-            obj.setAttributeRAW('voltage0', 'raw', num2str(value), true, obj.SRayCtrlDevices);
+            obj.setAttributeRAW('voltage0', 'raw', num2str(value), true, obj.SRayCtrlDevice);
         end
     end
     
     methods (Hidden, Access = protected)
         function setupInit(obj)
-            numDevs = obj.iio_context_get_devices_count(obj.iioCtx);
-            for dn = 1:length(obj.SRayCtrlDeviceNames)
-                for k = 1:numDevs
-                    devPtr = obj.iio_context_get_device(obj.iioCtx, k-1);
-                    name = obj.iio_device_get_name(devPtr);
-                    if strcmpi(obj.SRayCtrlDeviceNames{dn},name)
-                        attr = obj.iio_device_get_attr(devPtr,0);
-                        if strcmpi(attr,'label')
-                            val = obj.getDeviceAttributeRAW(attr,128,devPtr);
-                            if strcmpi(val, obj.SRayCtrlDevLabel)
-                                obj.SRayCtrlDevices = devPtr;
-                                break;
-                            else
-                                continue;
-                            end
-                        end
-                    end
-                end
-                if isempty(obj.SRayCtrlDevices)
-                   error('%s not found',obj.SRayCtrlDeviceNames{dn});
-                end
+            obj.SRayCtrlDevice = obj.getDev(obj.SRayCtrlDeviceName);
+            if isempty(obj.SRayCtrlDevice)
+               error('%s not found',obj.SRayCtrlDeviceName);
             end
```

Comment:
If using getDev this check isn't necessary since its built-in to getDev

Reply:
Fixed.